third_party/move/move-vm/runtime/src/data_cache.rs

```diff
@@ -27,19 +27,105 @@ use move_vm_types::{
 };
 use std::collections::btree_map::{BTreeMap, Entry};
 
+enum LazyDeserializationValue {
+    Serialized(Option<Bytes>),
```

Comment:
As discussed, we should cache size here otherwise we might not invalidate these bytes :)